### PR TITLE
Solution to issue: Replace "Hudson" by "Jenkins" in the codebase

### LINKED ISSUE #375: Replace "Hudson" by "Jenkins" in the codebase
There is some old Jenkinsfile Runner code which was originally a part of Jenkins test frameworks. This code uses the "Hudson" name though Jenkins was renamed to Hudson in 2011. It would be great to cleanup this code. Some occurences: https://github.com/jenkinsci/jenkinsfile-runner/search?q=Hudson

Acceptance criteria:

* "Hudson" occurrences in the documentation, Javadoc and comments are replaced with Jenkins
* "Hudson" occurrences in class and method names are replaced by "Jenkins". In this repository binary compatibility is not a concern even for public classes

### PR DESCRIPTION
**Pull request by darkphoenix17**
I have made all the changes required.
Please run and check it.
Also, I am just new to open source so if you find any error then can you please elaborate me the ways to solve it.

### CODE REVIEW

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Hi @darkphoenix17 . Please do not modify imports from dependencies, the code won't compile. We cannot do much about `hudson` in imports, because they are a part of public API of Jenkins. `hudson.model.hudson` is one of the only exceptions, it can be replaced by `jenkins.model.Jenkins`

**Comment by darkphoenix17** [comment]
> Hi @darkphoenix17 . Please do not modify imports from dependencies, the code won't compile. We cannot do much about `hudson` in imports, because they are a part of public API of Jenkins. `hudson.model.hudson` is one of the only exceptions, it can be replaced by `jenkins.model.Jenkins`

i am still tryin to fix it.....can you point out the specific area to work upon.

File: app/pom.xml [hunk 1]
```diff
@@ -35,7 +35,7 @@
        </plugin>
       <!-- TODO(oleg_nenashev): Standalone JAR generation works by luck: https://github.com/jenkinsci/jenkinsfile-runner/issues/350
            META-INF definitions are not merged correctly, and extension annotations from the core might be overridden by plugins.
-           Custom container description handlers are required to handle META-INF/annotations/hudson.Extension though it might be replaced by META-INF/annotations/hudson.Extension.txt.
+           Custom container description handlers are required to handle META-INF/annotations/jenkins.Extension though it might be replaced by META-INF/annotations/jenkins.Extension.txt.
```
Comment (oleg-nenashev) [inline app/pom.xml:38]: Also a part of API

File: bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java [hunk 2]
```diff
@@ -239,7 +239,8 @@ private String getMininumJenkinsVersion() throws IOException {
     }
 
     private boolean isVersionSupported() throws IOException {
-        return new VersionNumber(this.version).isNewerThanOrEqualTo(new VersionNumber(this.getMininumJenkinsVersion()));
+        return new hudson.util.VersionNumber(this.version)
```
Comment (oleg-nenashev) [inline bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java:246]: Can be reverted after you reverted the import call

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
There is an issue with API:

```
[2020-09-30T22:51:48.068Z] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project setup: Compilation failure

[2020-09-30T22:51:48.068Z] [ERROR] /home/jenkins/platform_linux_PR-376_3/setup/src/main/java/io/jenkins/jenkinsfile/runner/JenkinsEmbedder.java:[323,20] Jenkins(java.io.File,javax.servlet.ServletContext,hudson.PluginManager) has protected access in jenkins.model.Jenkins

[2020-09-30T22:51:48.068Z] [ERROR] -> [Help 1]
```

Both constructors are protected according to https://javadoc.jenkins-ci.org/jenkins/model/Jenkins.html , so you cannot use the class directly. It is possible to create a new class extending Jenkins, or just revert ` return new Hudson(home, webServer, getPluginManager());` for now.

Ideally we should make constructors public inside the Jenkins class, but it would be a longer discussion and a patch on the Jenkins core side

**Comment by PierreBtz** [comment]
fixes #375

File: bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java (hunk 2 above)
Comment (darkphoenix17) [reply bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java:246]: I made the changes again...I am getting errors again.
I am sorry if I am using your time for such thing but now where is the fault can you guide.

**Comment by oleg-nenashev** [comment]
@darkphoenix17 You have not addressed, hence the build fails with the same result as before the last commit: https://github.com/jenkinsci/jenkinsfile-runner/pull/376#pullrequestreview-500186734

**Comment by darkphoenix17** [comment]
> 
> 
> @darkphoenix17 You have not addressed, hence the build fails with the same result as before the last commit: [#376 (review)](https://github.com/jenkinsci/jenkinsfile-runner/pull/376#pullrequestreview-500186734)

I am still not getting it.
Can you guide me through steps to resolve this.

File: setup/src/main/java/io/jenkins/jenkinsfile/runner/JenkinsLauncher.java [hunk 3]
```diff
@@ -3,8 +3,12 @@
 import hudson.ClassicPluginStrategy;
 import hudson.util.PluginServletFilter;
 import io.jenkins.jenkinsfile.runner.bootstrap.Bootstrap;
-import io.jenkins.jenkinsfile.runner.util.HudsonHomeLoader;
+// <<<<<<< master
```
Comment (oleg-nenashev) [inline setup/src/main/java/io/jenkins/jenkinsfile/runner/JenkinsLauncher.java:6]: Needs fix

File: setup/src/main/java/io/jenkins/jenkinsfile/runner/JenkinsEmbedder.java [hunk 4]
```diff
@@ -322,7 +322,7 @@ protected Hudson newHudson() throws Exception {
             r.decorateHome(this, home);
         }
         try {
-            return new Hudson(home, webServer, getPluginManager());
+            return new Jenkins(home, webServer, getPluginManager());
```
Comment (oleg-nenashev) [inline setup/src/main/java/io/jenkins/jenkinsfile/runner/JenkinsEmbedder.java:325]:
```suggestion
            return new Hudson(home, webServer, getPluginManager());
```

This causes compilation failure

Comment (darkphoenix17) [reply setup/src/main/java/io/jenkins/jenkinsfile/runner/JenkinsEmbedder.java:325]: I am starting a clean slate now and checking to all the errors that I made and resolving it.

File: setup/src/main/java/io/jenkins/jenkinsfile/runner/JenkinsLauncher.java (hunk 3 above)
Comment (darkphoenix17) [reply setup/src/main/java/io/jenkins/jenkinsfile/runner/JenkinsLauncher.java:6]: Should the HudsonhomeLoader file name not be changed along with it's class usage in the other files?

**Comment by darkphoenix17** [comment]
hey @oleg-nenashev !! I have made another PR and solved this issue it seems as the checker has passed. You may review it here.
The title is "Solved Replace Issue"

**Comment by oleg-nenashev** [comment]
Closing in favor of #378

**Pull request closed without merging**